frontend/src/containers/pages/PlayScenarioPage/PlayScenarioPageMulti.jsx

```diff
@@ -44,43 +44,34 @@ export default function PlayScenarioPageMulti({ group }) {
   const [resourcesOpen, setResourcesOpen] = useState(false);
   const [noteOpen, setNoteOpen] = useState(false);
   const [previous, setPrevious] = useState(null);
-  const [error, setError] = useState(null);
 
-  // TODO: move these somewhere else ?
-  if (loading) return <LoadingPage text="Loading Scene..." />;
-  if (authError) return <></>;
+  const setError = useCallback((error) => {
+    if (!error) return;
+    if (error.status === 409) {
+      history.push(`/play/${scenarioId}/desync`);
+    } else if (error.status === 403) {
+      const roles = JSON.stringify(error.meta.roles_with_access);
+      history.push(`/play/${scenarioId}/invalid-role?roles=${roles}`);
+    } else {
+      history.push(`/play/${scenarioId}/error`);
+    }
+  }, []);
 
   useEffect(() => {
     const onSceneChange = async () => {
       if (sceneId && !previous) return;
-      const res = await navigate(user, group._id, previous, sceneId).catch(
-        (e) => setError(e?.response)
-      );
-      if (!sceneId) history.replace(`/play/${scenarioId}/multiplayer/${res}`);
+      if (sceneCache.get(sceneId)?.error) setError(sceneCache.get(sceneId));
+      try {
+        const res = await navigate(user, group._id, previous, sceneId);
```

Comment:
nit: naming, should say its the new `sceneId`

Reply:
its kind of implied? after all sceneId is something that's being used everywhere to refer to the current scene.

Reply:
I had to look up the definition of `navigate` to figure out what it does, because res (for my first pass of this code) was unclear.

Reply:
oh my bad I thought you were talking about something else. Yeah res is pretty bad lol.